### ISSUE TITLE
UnmarshalJSON returns error because of case sensitive tag fields

### ISSUE THREAD

**Issue by fdymylja** [golang/go#30124]
### What version of Go are you using (`go version`)?

<pre>
$ go version go1.11.5 windows/amd64
</pre>

### Does this issue reproduce with the latest release?
Yes

### What operating system and processor architecture are you using (`go env`)?
Windows 10
<details><summary><code>go env</code> Output</summary><br><pre>
$ go env
go env
set GOARCH=amd64
set GOBIN=
set GOCACHE=C:\Users\User\AppData\Local\go-build
set GOEXE=.exe
set GOFLAGS=
set GOHOSTARCH=amd64
set GOHOSTOS=windows
set GOOS=windows
set GOPATH=C:\Users\User\go
set GOPROXY=
set GORACE=
set GOROOT=C:\Go
set GOTMPDIR=
set GOTOOLDIR=C:\Go\pkg\tool\windows_amd64
set GCCGO=gccgo
set CC=gcc
set CXX=g++
set CGO_ENABLED=1
set GOMOD=
set CGO_CFLAGS=-g -O2
set CGO_CPPFLAGS=
set CGO_CXXFLAGS=-g -O2
set CGO_FFLAGS=-g -O2
set CGO_LDFLAGS=-g -O2
set PKG_CONFIG=pkg-config
set GOGCCFLAGS=-m64 -mthreads -fmessage-length=0 -fdebug-prefix-map=C:\Users\User\AppData\Local\Temp\go-build108285278=/tmp/go-build -gno-record-gcc-switches
</pre></details>

### What did you do?

Code on go playground: https://play.golang.org/p/uALETwdJpgC
Basically when having a json with different fields, one being upper case the other being lower case, json returns an error : "cannot unmarshal number into Go struct field Crashing.e of type string"
### What did you expect to see?

I expected to see the Unmarshal function to run correctly, the Crashing struct is still unmarshaled properly

### What did you see instead?

It unmarshaled it and encountered the issue (probably) and then  stopped. 

Adding one upper case field and one lower case field to the unmarshaling struct seems to fix the problem.

**Comment by mvdan** [comment]
Duplicate of https://github.com/golang/go/issues/14750.
